### ISSUE TITLE
Idea: Sequenced Moodles

### ISSUE THREAD

**Issue by Spectral-Fires** [kawaii/Moodles#48]
A feature of regular buffs/debuffs that I think would be a good addition from the game would be a way to make a way to have sequenced moodles. Think of some mechanics in extremes and savages where, when one debuff expires, it applies another related buff/debuff in it's place.

There's a couple different ways I could imagine implementing such a feature, but I think the simplest would be to just add a condition to the creation form along the lines of "On Removal" that will trigger when the duration expires (if temporary), or optionally when the original moodle is dispelled. 

Obviously there would need to be some restrictions and fail-safes built in to prevent issues or abuse, like I would almost certainly say that the option for cyclical variables would need to be disabled. My thought on this would be to require checking a box that adds a "conditional" flag to the moodle. Only moodles with the conditional flag could be automatically added by the removal of other moodles, and conditional moodles themselves could not, in turn, apply other moodles on their removal. The implementation in the game only ever allows an initial application to apply a secondary application, so I see no reason that such a feature should extend beyond the game's implementation of such a system.

Possible concerns with implementation aside, though, I do think having an analogue to the in-game system of sequenced buffs would be a neat opportunity to expand roleplay uses with the plugin.

**Comment by Glyceri** [comment]
I have gotten multiple requests for this feature, it shall be worked on!

**Comment by Glyceri** [comment]
So I got the feature working, but @kawaii as cute as he is wrote absolute spaghetti code. So I feel almost the only way this is possible is if you dispell the Moodle in ANY way, it'll apply the next Moodle. I will look further into this.

**Comment by Spectral-Fires** [comment]
Totally appreciate you looking into this! If I could even begin to competently write C# I'd be glad to try to branch and play around with it myself to untangle the spaghetti, but it is one of the few coding languages I just stare at and go "yep that's code alright."

**Comment by Glyceri** [comment]
This is added as of 1.0.0.32 in a very crude/entry level way. Let me know if this causes issues!

**Comment by Spectral-Fires** [comment]
A cursory mess-around with the functionality on my own seems to work alright. I'll put it through some more rigorous testing with other people after Thanksgiving and see if it causes any weird problems.

I can say the UI is bugged, only visually, looks like a simple fix, the row name/description is just nested in the dropdown if statement, rather than in a separate, preceeding TableNextColumn.
![image](https://github.com/user-attachments/assets/5f5a8aea-ba30-4f76-bd61-ddb14faf43c2)

Gonna try to resolve it here for my own practice, but like I said C# is far from my strong point here. Just pasting a snippet of the relevant bit.

```C#
                ImGui.TableNextColumn();
                ImGuiEx.TextV($"Apply on Dispell:");
                ImGuiEx.HelpMarker("The selected Moodle gets applied automatically upon ANY dispell of the current Moodle.");

                ImGui.TableNextColumn();
                ImGuiEx.SetNextItemFullWidth();
                
                if (ImGui.BeginCombo("##addnew", "Apply Moodle On Dispell..."))
                {
                    
                    ImGui.InputTextWithHint("##search", "Filter", ref Filter, 50);
                    if (ImGui.Selectable($"Clear", false, ImGuiSelectableFlags.None))
                    {
                        Selected.StatusOnDispell = Guid.Empty;
                        P.IPCProcessor.StatusModified(Selected.GUID);
                    }
```

**Comment by Glyceri** [comment]
Yeah idk why the ImGui.TableNextColumn(); made it show up in the first column still anyways. But if you remove the ImGuiEx.TextV it starts bugging out completely, so I just did this and put the selected dispell moodle on the right and told Kite to fix it in februari when he's back from being cringe.

**Comment by Spectral-Fires** [comment]
Gotta love UI. It's cursed at the best of times. XD

**Comment by Glyceri** [comment]
<img width="452" alt="image" src="https://github.com/user-attachments/assets/6ea320d5-fd2b-4281-b5cb-56790abed2bc">
Nvm fixed it. There will be no more icon, but beggers cannot be choosers

**Comment by Glyceri** [comment]
You know, Kite always tells me I can push as many updates a day as I want. But this will be the last for now. Fixed in 1.0.0.33
